### ISSUE TITLE
Please give sha hash for AppImage also to verify downloaded package.

### ISSUE THREAD

**Issue by Nokia808** [qpdf/qpdf#301]
As title saying, currently I do not see sha hash for  AppImage. It should be provided to verified downloaded package to avoid man-in-middle attacks.

Also, your signature is available in same file, but where is your public key? If I'm not wrong, to verify signature for downloaded package we need 3 files:

- downloaded software
- signature file
- public PGP key

And we need finger print of public PGP key to verify it before being imported.

I suggest to make public key available for download in this site as .asc file together with sig. file & demonstrate finger print for public key. This way can sure 100% that downloaded package is safe.

**Comment by jberkenbilt** [comment]
You are right, some of the signatures are missing. I must have had some problem in my release script that I didn't catch. I will fix and re-upload the signatures. You can find my public key [here](https://q.ql.org/pubkey.asc). I will make sure there is documentation of where you can find it. I thought it was in the README but I have not checked just now.

**Comment by jberkenbilt** [comment]
I can't fix this minute but will try to get to it later today. These are the checksums of my archived release. I will sign and upload ASAP.

```
f30ab95b28c3a78c5a8dc658f4ea6704d79a92732c05ccf1dcd3744c4197f41417dec3c5ca79f819de59f899461afc1220de6d4f46659a63c87fa3e8095e654b  qpdf-8.4.0-bin-mingw32.zip
3d30539e61ee1a2a4814f8b082ebd93eab687fc2b9a6fc989dc1bf1e4f4bdacd01d9aa864e7194e4578178db39d5236820644e14bef815c67dc0d4794276c0d5  qpdf-8.4.0-bin-mingw64.zip
0abc5f934c1d38846071a4df70807ec13a43ecf076c9512aa112a1fe431250e359de339cf4b2801a973f63378d9d9a3b76cc537cd2f9e8bf3e10260969d1c3bd  qpdf-8.4.0-bin-msvc32.zip
8f613a21e151ba6da85fae7ccf9a01f06211872fa3392dffc9abd90a7af432da4aec58d44c6db514563b74dfa045f189e6423c94a63464b28c6e144e5e4f7b82  qpdf-8.4.0-bin-msvc64.zip
13225c6285311e8fe89970971b082b9613e70d474ee71dd5572f1ae268234d366c2c9c7bf427c4758a746a1a27ecc0127d31044ab501dbc08885c8aa0a514028  qpdf-8.4.0.tar.gz
760d78b893002dad9dbd7d97c6ab2cfba44032b463c9da7e78a6df5d0f3ba699b9c5f4da9ef3cacbe5d23e8b23394db36ff877e180b581ca22ee253d16088318  qpdf-8.4.0-x86_64.AppImage
1c51e0a57e6d09e754185ba4329685d813caba3ebb91be91abe19bb549445b559255e7712aa0e08d1cc58c98709c3f9ea7be3e553563bb78a6a09b6784046c01  qpdf-8.4.0-x86_64.AppImage.zsync
```

**Comment by Nokia808** [comment]
@jberkenbilt 
Thank you very much ! This is exactly what I expect from you ! In fact beyond !
Do you know why I opened such issue ? Currently I have 2 portable programs on my system as AppImages: qpdf.ai & a messenger called Rambox. I checked Rambox hash & it is NOT as it given by developer on GitHub !!! For that I opened this issue to verify your application.

I used sha512sum command it it generated exactly same hash you provided (you did not mention that it is 512).

It seem that I'm a victum for man-in-middle attack regarding Rambox. I will opened an issue on their site urgently !

One thing: you did not mention finger print of your public key.

It seem that this issue is very important in these days ! My Rambox AppImage is undermined ! I thought before your replay that it's hash differ because I changed the permissions after download. But after I checked your application AppImage (which I already changed it's permissions to make it run) it seem that there is something wrong: either I'm affected by man-in-middle attack or the developer of Rambox make mistake !!

**Comment by Nokia808** [comment]
@jberkenbilt 
Please kindly see discussion on following link that I opened in Rambox if it of usefulness to you:
https://github.com/ramboxapp/community-edition/issues/2109

**Comment by jberkenbilt** [comment]
I updated the signed checksum files in the 8.4.0 release. So silly....I used to use zsh, but I had decided to use bash for a while to get my bash environment up to date with my zsh environment, and I hadn't replaced `for i in $files` with `for i in ${files[*]}` in some commands that I cut and paste instead of running from a script.

I added a note to the TODO list to make sure that it's easier to find my public key. I'll make sure that's updated for the next release. I can't retroactively change it like I can for the checksum files.

Thanks again for noticing this. It's interesting that no one else has noticed.
